### PR TITLE
feat(NMT): clients

### PR DESCRIPTION
**Pull request by junkin**
NMT text-text, speech-text, speech-speech clients

### CODE REVIEW

File: riva/clients/nmt/riva_asr_client_helper.cc [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+bool
+OpenAudioDevice(
+    const char* devicename, snd_pcm_t** handle, snd_pcm_stream_t stream_type, int channels,
+    int rate, unsigned int latency)
+{
+  int rc;
+  static snd_output_t* log;
+
+  std::cerr << "latency " << latency << std::endl;
```
Comment (drnikolaev) [inline riva/clients/nmt/riva_asr_client_helper.cc]: turn to LOG() pls

File: riva/clients/nmt/riva_asr_client_helper.cc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+  if ((rc = snd_pcm_open(handle, devicename, stream_type, 0)) < 0) {
+    printf("unable to open pcm device for recording: %s\n", snd_strerror(rc));
```
Comment (drnikolaev) [inline riva/clients/nmt/riva_asr_client_helper.cc]: turn to LOG() pls

File: riva/clients/nmt/riva_nmt.cc [hunk 3]
```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
```
Comment (drnikolaev) [inline riva/clients/nmt/riva_nmt.cc]: we probably should be consistent with this copyright coimment

File: riva/clients/nmt/riva_nmt.cc [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+      }
+      current_line++;
+    }
+  }
+}
+
+std::vector<double>*
+percentiles(std::vector<double> v)
+{
+  std::vector<double>* results = new std::vector<double>();
```
Comment (drnikolaev) [inline riva/clients/nmt/riva_nmt.cc]: this is really against the modern C++ spirit :)

Comment (junkin) [reply riva/clients/nmt/riva_nmt.cc]: oof.  that's copy pasta from https://github.com/nvidia-riva/cpp-clients/blob/main/riva/clients/tts/riva_tts_perf_client.cc#L163

File: riva/clients/nmt/riva_asr_client_helper.cc (hunk 1 above)
Comment (junkin) [reply riva/clients/nmt/riva_asr_client_helper.cc]: refactored away

File: riva/clients/nmt/riva_asr_client_helper.cc (hunk 2 above)
Comment (junkin) [reply riva/clients/nmt/riva_asr_client_helper.cc]: refactored away

File: riva/clients/nmt/riva_nmt.cc (hunk 3 above)
Comment (junkin) [reply riva/clients/nmt/riva_nmt.cc]: yes - done

**Review by drnikolaev — APPROVED** [review]

**Pull request merged**